File: META_Moez/Metastone/src/tn/edu/esprit/model/Utilisateur.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.edu.esprit.model;


/**
 *
 * @author admin
 */
public class Utilisateur {
    private int ID_UTILISATEUR;
    private String date_naiss;
    private String mtc,nom,prenom,email,tel,adresse,nom_utilisateur,
            mot_de_passe,sexe,photo_de_profil,role,rank,id,id_consommateur;
    private int code;
   
       
//constructor
    public Utilisateur(String nom, String prenom, String date_naiss,String email,String tel,String adresse,String nom_utilisateur,String mot_de_passe,String sexe,String photo_de_profil,String mtc,String role, String rank,String id_consommateur) {
        this.mtc = mtc;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naiss = date_naiss;
        this.email = email;
        this.tel = tel;
        this.adresse = adresse;
        this.nom_utilisateur = nom_utilisateur;
        this.mot_de_passe = mot_de_passe;
        this.sexe = sexe;
        this.photo_de_profil = photo_de_profil;
        this.role = role;
        this.rank = rank;
        this.id_consommateur=id_consommateur;
    }
    public Utilisateur(int ID_UTILISATEUR, String nom, String prenom, String date_naiss, String email, String tel, String adresse, String nom_utilisateur, String mot_de_passe, String sexe, String photo_de_profil,String mtc, String role, String rank, String id_consommateur) {
        this.ID_UTILISATEUR = ID_UTILISATEUR;
        this.mtc = mtc;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naiss = date_naiss;
        this.email = email;
        this.tel = tel;
        this.adresse = adresse;
        this.nom_utilisateur = nom_utilisateur;
        this.mot_de_passe = mot_de_passe;
        this.sexe = sexe;
        this.photo_de_profil = photo_de_profil;
        this.role = role;
        this.rank = rank;
        this.id_consommateur = id_consommateur;   
    }

    //..
    public Utilisateur ()
    {
       mtc="";
       nom="";
       prenom="";
       date_naiss="";
       email="";
       tel="";
       adresse="";
       nom_utilisateur="";
       mot_de_passe="";
       sexe="";
       photo_de_profil="";
       role="";
       rank="";
       id_consommateur="";
       
       
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    




    public String getId() {
        return id;
    }

    public String getMtc() {
        return mtc;
    }

    public String getNom() {
        return nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public String getDate_naiss() {
        return date_naiss;
    }

    public String getEmail() {
        return email;
    }

    public String getTel() {
        return tel;
    }

    public String getAdresse() {
        return adresse;
    }

    public String getNom_utilisateur() {
        return nom_utilisateur;
    }

    public String getMot_de_passe() {
        return mot_de_passe;
    }

    public String getSexe() {
        return sexe;
    }

    public String getPhoto_de_profil() {
        return photo_de_profil;
    }

    public String getRole() {
        return role;
    }

    public String getRank() {
        return rank;
    }

    public String getId_consommateur() {
        return id_consommateur;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setMtc(String mtc) {
        this.mtc = mtc;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public void setDate_naiss(String date_naiss) {
        this.date_naiss = date_naiss;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setTel(String tel) {
        this.tel = tel;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public void setNom_utilisateur(String nom_utilisateur) {
        this.nom_utilisateur = nom_utilisateur;
    }

    public void setMot_de_passe(String mot_de_passe) {
        this.mot_de_passe = mot_de_passe;
    }

    public void setSexe(String sexe) {
        this.sexe = sexe;
    }

    public void setPhoto_de_profil(String photo_de_profil) {
        this.photo_de_profil = photo_de_profil;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public void setRank(String rank) {
        this.rank = rank;
    }

    public void setId_consommateur(String id_consommateur) {
        this.id_consommateur = id_consommateur;
    }

    public int getID_UTILISATEUR() {
        return ID_UTILISATEUR;
    }

    public void setID_UTILISATEUR(int ID_UTILISATEUR) {
        this.ID_UTILISATEUR = ID_UTILISATEUR;
    }
    
    
    //Affichage

    @Override
    public String toString() {
        return "Utilisateur{" + "ID_UTILISATEUR=" + ID_UTILISATEUR + ", date_naiss=" + date_naiss + ", mtc=" + mtc + ", nom=" + nom + ", prenom=" + prenom + ", email=" + email + ", tel=" + tel + ", adresse=" + adresse + ", nom_utilisateur=" + nom_utilisateur + ", mot_de_passe=" + mot_de_passe + ", sexe=" + sexe + ", photo_de_profil=" + photo_de_profil + ", role=" + role + ", rank=" + rank + ", id=" + id + ", id_consommateur=" + id_consommateur + ", code=" + code + '}';
    }

   
    
    
}

File: META_Moez/Metastone_web/src/Entity/Utilisateur.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Utilisateur
 *
 * @ORM\Table(name="utilisateur", uniqueConstraints={@ORM\UniqueConstraint(name="Nom_utilisateur", columns={"Nom_utilisateur"}), @ORM\UniqueConstraint(name="E_mail", columns={"email"})})
 * @ORM\Entity
 */
class Utilisateur
{
    /**
     * @var int
     *
     * @ORM\Column(name="ID_utilisateur", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idUtilisateur;

    /**
     * @var string
     *
     * @ORM\Column(name="Nom", type="string", length=255, nullable=false)
     */
    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="Prenom", type="string", length=255, nullable=false)
     */
    private $prenom;

    /**
     * @var string
     *
     * @ORM\Column(name="age", type="string", length=22, nullable=false)
     */
    private $age;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=255, nullable=false)
     */
    private $email;

    /**
     * @var int
     *
     * @ORM\Column(name="Tel", type="integer", nullable=false)
     */
    private $tel;

    /**
     * @var string
     *
     * @ORM\Column(name="Adresse", type="string", length=40, nullable=false)
     */
    private $adresse;

    /**
     * @var string
     *
     * @ORM\Column(name="Nom_utilisateur", type="string", length=20, nullable=false)
     */
    private $nomUtilisateur;

    /**
     * @var string
     *
     * @ORM\Column(name="Mot_de_passe", type="string", length=50, nullable=false)
     */
    private $motDePasse;

    /**
     * @var string
     *
     * @ORM\Column(name="Sexe", type="string", length=255, nullable=false)
     */
    private $sexe;

    /**
     * @var string
     *
     * @ORM\Column(name="Photo_de_profil", type="string", length=80, nullable=false)
     */
    private $photoDeProfil;

    /**
     * @var int
     *
     * @ORM\Column(name="MTC", type="integer", nullable=false)
     */
    private $mtc;

    /**
     * @var string
     *
     * @ORM\Column(name="Role", type="string", length=255, nullable=false)
     */
    private $role;

    /**
     * @var string
     *
     * @ORM\Column(name="Rank", type="string", length=255, nullable=false)
     */
    private $rank;

    /**
     * @var string
     *
     * @ORM\Column(name="Id_consommateur", type="string", length=255, nullable=false)
     */
    private $idConsommateur;

    /**
     * @var int|null
     *
     * @ORM\Column(name="code", type="integer", nullable=true)
     */
    private $code;

    /**
     * @var string
     *
     * @ORM\Column(name="etat", type="string", length=255, nullable=false, options={"default"="activer"})
     */
    private $etat = 'activer';

    public function getIdUtilisateur(): ?int
    {
        return $this->idUtilisateur;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getAge(): ?string
    {
        return $this->age;
    }

    public function setAge(string $age): self
    {
        $this->age = $age;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getTel(): ?int
    {
        return $this->tel;
    }

    public function setTel(int $tel): self
    {
        $this->tel = $tel;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getNomUtilisateur(): ?string
    {
        return $this->nomUtilisateur;
    }

    public function setNomUtilisateur(string $nomUtilisateur): self
    {
        $this->nomUtilisateur = $nomUtilisateur;

        return $this;
    }

    public function getMotDePasse(): ?string
    {
        return $this->motDePasse;
    }

    public function setMotDePasse(string $motDePasse): self
    {
        $this->motDePasse = $motDePasse;

        return $this;
    }

    public function getSexe(): ?string
    {
        return $this->sexe;
    }

    public function setSexe(string $sexe): self
    {
        $this->sexe = $sexe;

        return $this;
    }

    public function getPhotoDeProfil(): ?string
    {
        return $this->photoDeProfil;
    }

    public function setPhotoDeProfil(string $photoDeProfil): self
    {
        $this->photoDeProfil = $photoDeProfil;

        return $this;
    }

    public function getMtc(): ?int
    {
        return $this->mtc;
    }

    public function setMtc(int $mtc): self
    {
        $this->mtc = $mtc;

        return $this;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function setRole(string $role): self
    {
        $this->role = $role;

        return $this;
    }

    public function getRank(): ?string
    {
        return $this->rank;
    }

    public function setRank(string $rank): self
    {
        $this->rank = $rank;

        return $this;
    }

    public function getIdConsommateur(): ?string
    {
        return $this->idConsommateur;
    }

    public function setIdConsommateur(string $idConsommateur): self
    {
        $this->idConsommateur = $idConsommateur;

        return $this;
    }

    public function getCode(): ?int
    {
        return $this->code;
    }

    public function setCode(?int $code): self
    {
        $this->code = $code;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }


}
